patches/api/0418-Expose-more-native-map-methods-for-PDC.patch

```diff
@@ -0,0 +1,281 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: RyGuy <oboylery@gmail.com>
+Date: Tue, 28 Mar 2023 01:27:28 -0400
+Subject: [PATCH] Expose more native map methods for PDC
+
+This commit has PDC inherit Iterable<NamespacedKey>, and implements these
+methods
+- #setIfAbsent(NamespacedKey, PersistentDataType, Object)
+- #computeIfAbsent(NamespacedKey, PersistentDataType, Function)
+- #computeIfPresent(NamespacedKey, PersistentDataType, BiFunction)
+- #compute(NamespacedKey, PersistentDataType, BiFunction)
+- #getOrThrow(NamespacedKey, PersistentDataType, Supplier)
+- #getOrThrow(NamespacedKey, PersistentDataType)
+- #replace(NamespacedKey, PersistentDataType, Object, Object)
+- #replace(NamespacedKey, PersistentDataType, Object)
+
+diff --git a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+index d30ad7cae1dfbb47ab1a6e3e2fcd6fb76163fe78..066314a9d4a53f98bc4cdf2bd1ace63d91e2560e 100644
+--- a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
++++ b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+@@ -1,15 +1,22 @@
+ package org.bukkit.persistence;
+ 
+-import java.util.Set;
+ import org.bukkit.NamespacedKey;
+ import org.jetbrains.annotations.NotNull;
+ import org.jetbrains.annotations.Nullable;
+ 
++import java.util.Iterator;
++import java.util.NoSuchElementException;
++import java.util.Objects;
++import java.util.Set;
++import java.util.function.BiFunction;
++import java.util.function.Function;
++import java.util.function.Supplier;
++
+ /**
+  * This interface represents a map like object, capable of storing custom tags
+  * in it.
+  */
+-public interface PersistentDataContainer {
++public interface PersistentDataContainer extends Iterable<NamespacedKey> {
```

Comment:
IDE is screaming at me because this would require changing `#getKeys` to use Key
Wasn't sure if this should be done because it can be considered an API break?

Reply:
yeah, don't change the return type of getKeys(), but what we can do is add a new method, `keySet()` (also following Map method names) that returns a `Set<Key>`. And then use that to get the iterator.